include/nbl/asset/IDescriptorSet.h

```diff
@@ -127,7 +134,7 @@ class NBL_API IDescriptorSet : public virtual core::IReferenceCounted
 				{
 					if (desc != desc)
 						return true;
-					return buffer != other.buffer;
+					return info.buffer != other.info.buffer;
 				}
 		};
 
```

Comment:
I think we can move `SWriteDescriptorSet` and `SCopyDescriptorSet` to `IGPUDescriptorSet` cause no one uses it outside of `video` namespace

Reply:
@achalpandeyy what do you think about moving `SWrite` and `SCopy` to `IGPUDescriptorSet` ?

Reply:
Yeah I think it can be done

Reply:
Only thing standing in the way is this https://github.com/Devsh-Graphics-Programming/Nabla/blob/9b5b2a443608d2fc1fe4da8af85947961cae4e0a/include/nbl/asset/utils/ICPUVirtualTexture.h#L531

Reply:
Did the Virtual Texturing example run before on the new API ?

if no you can `ifdef` this function out for now

if yes then refactor it to manipulate the `_dstSet` directly (since the infos are mutable and so on)

Reply:
The only usage of this function and other instances of `ICPUDescriptorSet::SWriteDescriptorSet` seems to be in `CMitsubaLoader.cpp`. The examples (20 and 41) only use `IGPUVirtualTexture::getDescriptorSetWrites` (which doesn't create any problems). So I was wondering if you could remove this function entirely?

Reply:
don't remove I'll need it but make it so that it writes into `_dstSet` directly when its `ICPUDescriptorSet` (also give it a new name)

> then refactor it to manipulate the _dstSet directly (since the infos are mutable and so on)
you can even validate a little bit if the ...Binding args match the layout!

Reply:
Did the modification to directly write info into `_dstSet`.
Extra validation is not really required or rather it is already being done just by calling `_dstSet->getDescriptorInfos(binding_numer)` because if the binding number doesn't exist in the layout then we get an empty `SRange`.

Reply:
Also moved `SWriteDescriptorSet` and `SCopyDescriptorSet` to `IGPUDescriptorSet`, so I think it can be resolved now.